app/models/pet.rb

```diff
@@ -12,4 +12,10 @@ def adoption_status_text
     end
   end
 
+  def approved_applicant
+    pet_applications.find do |application|
+      application.approval_status == "approved"
+    end
+  end
```

Comment:
This is used as a helper method in the pet model to remove the delete button if an application is approved.

Reply:
Great use of a helper method. I'm sure this could have been useful for other views where buttons appear/disappear based on certain criteria.